core/db/entities.py

```diff
@@ -1,19 +1,40 @@
 import uuid  # noqa
 
-from sqlalchemy import Column, String
+import sqlalchemy
 
 from core.db import db
 from core.db.types import GUID
 
 
+class Organisation(db.Model):
+    __tablename__ = "organisation_dim"
+
+    organisation_id = sqlalchemy.Column(GUID(), default=uuid.uuid4, primary_key=True)
+    organisation_name = sqlalchemy.Column(
+        sqlalchemy.String, nullable=False, unique=True
+    )
+    geography = sqlalchemy.Column(sqlalchemy.String, nullable=True)
+
+    contacts = sqlalchemy.orm.relationship("Contact", back_populates="organisation")
+
+
 class Contact(db.Model):
-    id = Column(
-        "id",
+    __tablename__ = "contact_dim"
+
+    contact_id = sqlalchemy.Column(
         GUID(),  # this should be UUIDType once using Postgres
         default=uuid.uuid4,
         primary_key=True,
     )
-    email_address = Column("email", String, nullable=False)
-    contact_name = Column("contact_name", String, nullable=True)
-    organisation = Column("organisation", String, nullable=True)
-    telephone = Column("telephone", String, nullable=True)
+    email_address = sqlalchemy.Column(sqlalchemy.String, nullable=False, unique=True)
+    contact_name = sqlalchemy.Column(sqlalchemy.String, nullable=True)
+    organisation_id = sqlalchemy.Column(
+        sqlalchemy.String,
+        sqlalchemy.ForeignKey("organisation_dim.organisation_id"),
+        nullable=False,
+    )
+    telephone = sqlalchemy.Column(sqlalchemy.String, nullable=True)
+    #
+    organisation = sqlalchemy.orm.relationship(
```

Comment:
Am a bit confused why we have both `organisation` and `organisation_id` here, do we know if we need both and how they map to underlying SQL?

Reply:
so, organisation here is the name of the relationship to the whole table, not a single field of it. This is standard naming convention when referencing the parent entity from a child. All the SQL is handled under the hood by SQL Alchemy.

So now in ORM style queries, we can access the organisation from any contact via attribute lookups i.e. Contact.organisation.geography  -> it makes attributes available for the parent, directly from the child. Does that make sense, I'm happy to demo if not